napari/_vispy/layers/surface.py

```diff
@@ -35,9 +59,7 @@ def _on_data_change(self):
         else:
             # Offsetting so pixels now centered
             # coerce to float to solve vispy/vispy#2007
-            vertices = np.asarray(
-                self.layer._data_view[:, ::-1], dtype=np.float32
```

Comment:
Most mesh software follows a convention on the order of vertices to determine whether the normal faces in or out. This `[:, ::-1]` Was reversing the order of the points, thus inverting all the normals. Does anyone know why this was introduced? To me, it looks like it might be a copy-paste leftover from the `Points` layer, where reversing is used to diplay the latest added point on top.

Reply:
Is this reversing the order of the vertices or swapping the dims order of the points? I think it might be the later. If so, I think this is because we store thing zyx in napari and vispy expects xyz.

Reply:
I am pretty sure the order of the vertices in a give faces is set by `self.layer._view_faces` below.

Reply:
Uh-oh, got confused! Thanks :)

Reply:
Should be fixed now: I reverse the faces definition as well, which should restore handedness after the `xyz` to `zyx` change.

Reply:
should be or is? :) if this is fixed we should be good to merge now!

Reply:
"should" as in "I hope I didn't break something else" :P

On a side note, I think we should wait for others to chip in before we merge; this PR even more than #3430 has a lot of new API, so there's probably some talk to have there. Also: should this PR include GUI access to this stuff?